typeclasses/npcshop/npcshop.py

```diff
@@ -0,0 +1,259 @@
+import inspect
+from evennia.utils import evmenu
+from evennia import Command
+from evennia import CmdSet
+from evennia import DefaultObject
+from typeclasses.rooms import Room
+from typeclasses.exits import Exit
+from typeclasses.items import Item
+from world.economy import format_coin as as_price
+from world.economy import transfer_funds, InsufficientFunds, value_to_coin
+from evennia.utils.create import create_object
+
+def get_wares(caller):
+    """
+    Gets items located in the designated storeroom of the caller's location with a price assigned
+    Only descendants of the Item typeclass are eligible for sale
+    """
+    return [ware for ware in caller.location.db.storeroom.contents if Item in inspect.getmro(ware.__class__) and ware.db.value]
```

Comment:
I would second Griatch's recommendation to implement this using the `inherits_from` function from the evennia.utils.utils module.